#!/bin/bash

CATEGORY=media-video
PN=parole
PATCH=parole-audio-order.patch

mkdir ${CATEGORY}/${PN}/files || exit 1
for i in ${CATEGORY}/${PN}/${PN}-0.8*.ebuild; do
	cat > ${PATCH} << EOF
--- a/${i}
+++ b/${i}
@@ -30,7 +30,8 @@
 	taglib? ( >=media-libs/taglib-1.6:0= )"
 RDEPEND="\${COMMON_DEPEND}
 	media-plugins/gst-plugins-meta:1.0"
+PATCHES=( "\${FILESDIR}"/${PATCH} )
 DEPEND="\${COMMON_DEPEND}
 	dev-util/intltool
 	sys-devel/gettext
 	virtual/pkgconfig
EOF
	git apply ${PATCH} || exit 1
	rm ${PATCH}
done
cat > ${CATEGORY}/${PN}/files/${PATCH} << EOF
--- parole-0.8.0/src/gst/parole-gst.c	2015-02-28 15:23:54.000000000 +0100
+++ parole-0.8.0/src/gst/parole-gst.c	2016-09-29 13:31:33.708857023 +0200
@@ -2895,11 +2895,9 @@
             }
         }
 
-        ret = g_list_reverse (ret);
-
         if ( gst->priv->use_custom_subtitles == TRUE )
         {
-            ret = g_list_prepend (ret, g_strdup_printf("%s",gst->priv->custom_subtitles));
+            ret = g_list_append (ret, g_strdup_printf("%s",gst->priv->custom_subtitles));
         }
     }
     else
@@ -2908,7 +2906,7 @@
         return NULL;
     }
 
-    return ret;
+    return g_list_reverse (ret);
 }
 
 gboolean
EOF
